src/shadow-embed.js

```diff
@@ -314,6 +316,44 @@ function getStylesheetRules(doc, css) {
   }
 }
 
+/**
+ * @param {!ShadowRoot} shadowRoot
+ * @param {string} name
+ * @param {string} cssText
+ */
+export function installShadowStyle(shadowRoot, name, cssText) {
+  const doc = shadowRoot.ownerDocument;
+  const win = toWin(doc.defaultView);
+  if (
+    shadowRoot.adoptedStyleSheets !== undefined &&
```

Comment:
What do you think about placing this conditional expression in the doc level, similar to what we did for display locking? We could also cache the value since support should not change across installations of many shadow roots on one document.

Reply:
IMHO the `shadowRoot.adoptedStyleSheets` is fast enough in this case and has all data needed, so I think the additional cache is not justified.